### PR TITLE
docs(cache): added docs for cache options - build dependencies

### LINKED ISSUE #3250: Document webpack change: add build dependencies
*A pull request by @sokra was merged and maintainers requested a documentation change.*

See pull request: https://github.com/webpack/webpack/pull/9567

---

**What kind of change does this PR introduce?**
feature

**Did you add tests for your changes?**
yes

**Does this PR introduce a breaking change?**
yes

**What needs to be documented once your changes are merged?**
* `cache.managedPaths` is an array of package-manager only managed paths. webpack will avoid hashing/timestamping them, assume the version is unique and will use this as snapshot. (for both memory and filesystem cache) Defaults to the `node_modules` webpack is inside of.
* `cache.buildDependencies` is an object of arrays of additional code dependencies of the build. webpack will use an hash of these items and all dependencies to invalidate the filesystem cache. `cache.buildDependencies.defaultWebpack` defaults to `webpack/lib` to get all dependencies of webpack.
* It's recommended to set `cache.buildDependencies.config: [__filename]` in `webpack.config.js` to get the config and all dependencies.

### PR DESCRIPTION
**Pull request by pranshuchittora**
Closes #3250 
_describe your changes..._

- [x] Read and sign the [CLA][1]. PRs that haven't signed it won't be accepted.
- [x] Make sure your PR complies with the [writer's guide][2].
- [ ] Review the diff carefully as sometimes this can reveal issues.
- __Remove these instructions from your PR as they are for your eyes only.__

[1]: https://cla.js.foundation/webpack/webpack.js.org
[2]: https://webpack.js.org/contribute/writers-guide/

### CODE REVIEW

File: src/content/configuration/other-options.md [hunk 1]
```diff
@@ -100,6 +101,29 @@ module.exports = {
 
 While setting `cache.type` to `filesystem` opens up more options for configuration.
 
+### `cache.buildDependencies`
+
+`Object: [path]`
+
+`cache.buildDependencies` is an object of arrays of additional code dependencies of the build. webpack will use an hash of these items and all dependencies to invalidate the filesystem cache.
```
Comment (jamesgeorge007) [inline src/content/configuration/other-options.md]:
```suggestion
`cache.buildDependencies` is an object of arrays of additional code dependencies of the build. webpack will use a hash of these items and all dependencies to invalidate the filesystem cache.
```

File: src/content/configuration/other-options.md [hunk 2]
```diff
@@ -100,6 +101,29 @@ module.exports = {
...
+
+Defaults to `webpack/lib` to get all dependencies of webpack.
+
+T> It's recommended to set `cache.buildDependencies.config: [__filename]` in `webpack.config.js` to get the config and all dependencies.
```
Comment (EugeneHlushko) [inline src/content/configuration/other-options.md]:
```suggestion
T> It's recommended to set `cache.buildDependencies.config: [__filename]` in your webpack configuration to get the latest configuration and all dependencies.
```

File: src/content/configuration/other-options.md [hunk 3]
```diff
@@ -100,6 +101,29 @@ module.exports = {
...
+
+```javascript
+module.exports = {
+  cache: {
+    buildDependencies: {
+      // recommended to invalidate cache on config changes
+      // This also makes all dependencies of this file build dependencies
```
Comment (EugeneHlushko) [inline src/content/configuration/other-options.md]:
```suggestion
      // This also makes all dependencies of this file - build dependencies
```

File: src/content/configuration/other-options.md [hunk 4]
```diff
@@ -100,6 +101,29 @@ module.exports = {
...
+
+Defaults to `webpack/lib` to get all dependencies of webpack.
+
+T> It's recommended to set `cache.buildDependencies.config: [__filename]` in `webpack.config.js` to get the config and all dependencies.
+
+```javascript
+module.exports = {
+  cache: {
+    buildDependencies: {
+      // recommended to invalidate cache on config changes
```
Comment (EugeneHlushko) [inline src/content/configuration/other-options.md]: Duplication of what you've said earlier in the W> block, maybe not necessary?

File: src/content/configuration/other-options.md [hunk 5]
```diff
@@ -144,6 +168,12 @@ module.exports = {
 };
 ```
 
+### `cache.managedPaths`
+
+`Array[path]`
+
+`cache.managedPaths` is an array of package-manager only managed paths. webpack will avoid hashing/timestamping them, assume the version is unique and will use this as snapshot. (for both memory and filesystem cache) Defaults to the `node_modules` webpack is inside of.
```
Comment (EugeneHlushko) [inline src/content/configuration/other-options.md]:
```suggestion
`cache.managedPaths` is an array of package-manager only managed paths. webpack will avoid hashing and timestamping them, assume the version is unique and will use it as a snapshot (for both memory and filesystem cache).
```

**Comment by EugeneHlushko** [comment]
THis got stalled, lets look at #3348 and #3661 if that doesnt work in chronological order

**Pull request closed without merging**